### ISSUE TITLE
Abstract away the binary classification task

### ISSUE THREAD

**Issue by MatthewJA** [chengsoonong/crowdastro#91]
Explicitly using a binary classifier and having all the training data in this format is leading to unnecessarily complex and hard to write code, making the data format really weird, and making the jump to other datasets harder. I should write some kind of `RGZClassifier` that has a scikit-learn interface but takes some representation of a Radio Galaxy Zoo subject instead of SWIRE objects and returns coordinates.

**Comment by MatthewJA** [comment]
We want to assign each radio object the location of the corresponding host galaxy. I think this is a good framing for the problem, since we want to implement into Radio Galaxy Zoo. I'll call the representation of these radio objects _subjects_ in analogy with Radio Galaxy Zoo.

A subject contains
- a RA/DEC location,
- a 5' × 5' radio image centred on the radio object,
- a name (e.g. `ATLAS3 J002933.7-440118C`), and
- a collection of nearby infrared objects, their pixel coordinates on the image, and their associated fluxes and stellarities.

This can be represented by 4 arrays. These can then be stacked per radio object. I'm not sure how to deal with the fact that there are different numbers of infrared objects per subject. I could probably store them as object arrays?

An `RGZClassifier` has a `predict` method that takes these arrays and returns the `(x, y)` pixel coordinates of the classification.

**Comment by MatthewJA** [comment]
#77 will be easier with these changes!

**Comment by MatthewJA** [comment]
Here's my initial attempt at formalising the problem.

Consider a radio object `r` that we want to label with an infrared object, and a set `X` of infrared objects. Let `y : X \to \{0, 1\}` represent whether a given `x \in X` is the infrared object that labels `r`. There exists exactly one `x \in X` such that `y(x) = 1`; for all other `x \in X`, `y(x) = 0`. Then we want to model `p(y(x) = 1 | x, r)`.

Is that on the right track?

**Comment by chengsoonong** [comment]
Looking good. You may want to relax the assumption that there is exactly one x\in X s.t. y=1. It may not buy you that much, and it is violated in some interesting cases.
